cmd/maintenance-manager/main.go

```diff
@@ -147,12 +152,24 @@ func main() {
 	ctx := ctrl.SetupSignalHandler()
 	mgrClient := mgr.GetClient()
 
+	ocpUtils, err := openshift.NewOpenshiftUtils(ctx, mgr.GetAPIReader())
+	if err != nil {
+		setupLog.Error(err, "unable to create openshift utils")
+		os.Exit(1)
+	}
+
+	if ocpUtils.IsOpenshift() {
+		setupLog.Info("openshift cluster detected",
+			"isOpenshift", ocpUtils.IsOpenshift(), "isHypershift", ocpUtils.IsHypershift())
+	}
+
 	if err = (&controller.NodeMaintenanceReconciler{
 		Client:                   mgrClient,
 		Scheme:                   mgr.GetScheme(),
 		CordonHandler:            cordon.NewCordonHandler(mgrClient, k8sInterface),
 		WaitPodCompletionHandler: podcompletion.NewPodCompletionHandler(mgrClient),
 		DrainManager:             drain.NewManager(ctrl.Log.WithName("DrainManager"), ctx, k8sInterface),
+		MCPManager:               openshift.NewMCPManager(ocpUtils, mgrClient),
```

Comment:
nit: just an opinion

IMO, It would be better to explicitly select the manager in `main.go` and skip initializing MCPManager for non-OpenShift clusters. This helps avoid confusion in the Reconciler code, where it's not immediately clear that certain logic applies only to OpenShift.

Reply:
so, you prefer to have :

```
if r.MCPManager != nil {
   // do stuff with MCPManager
}
```
?

i was kinda on the fence between the two approaches. lemme rework this then.

Reply:
I prefer to have an "if openshift" check everywhere to decide if we need to make a call to MCPManager or not. This means we will call openshift-specific logic very explicitly.

The current implementation is also ok, but for me it is a bit harder to follow. It is ok to keep everything as is  if you prefer so.